src/lib/smartlist_core/smartlist_foreach.h

```diff
@@ -83,6 +83,14 @@
          ++var ## _sl_idx) {                                    \
       var = (sl)->list[var ## _sl_idx];
 
+#define SMARTLIST_FOREACH_REVERSE_BEGIN(sl, type, var)  \
```

Comment:
I think this will be incompatible with SMARTLIST_DEL_CURRENT; we should maybe add a comment?

Reply:
A comment about this was added in a336d816a68e5eaddd9d80f7179699274b367a1d.